beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -855,6 +855,45 @@ where
         }
 
         let (req, substream) = substream;
+        let current_fork = self.fork_context.current_fork();
+        let spec = &self.fork_context.spec;
+
+        match &req {
+            RequestType::BlocksByRange(request) => {
+                let max_allowed = spec.max_request_blocks(current_fork) as u64;
+                if *request.count() > max_allowed {
+                    self.events_out.push(HandlerEvent::Err(HandlerErr::Inbound {
+                        id: self.current_inbound_substream_id,
+                        proto: Protocol::BlocksByRange,
+                        error: RPCError::InvalidData(format!(
+                            "requested exceeded limit. allowed: {}, requested: {}",
+                            max_allowed,
+                            request.count()
+                        )),
+                    }));
+                    return self.shutdown(None);
+                }
+            }
+            RequestType::BlobsByRange(request) => {
+                let max_requested_blobs = request
+                    .count
+                    .saturating_mul(spec.max_blobs_per_block_by_fork(current_fork));
+                let max_allowed = spec.max_request_blob_sidecars(current_fork) as u64;
+                if max_requested_blobs > max_allowed {
```

Comment:
Doing this check during decoding was nice because it protected the rate-limiter, which is currently vulnerable to overflow (not a big issue, because peer penalties kick in shortly after). I'll open a follow-up PR to fix that.

Reply:
yeah It was a suggestion from me to not do it during `protocol.rs` and the `Encoder` and `Decoder` as decode should only care about encoding and decoding not protocol logic, 
thanks for opening the issue Michael!

Reply:
This check in the handler is still before hitting the rate limiter though. The rate limiter check happens in the rpc behaviour on a `HandlerEvent::Ok`. We don't send a `HandlerEvent::Ok` if the check above doesn't pass